sources/osgUtil/DisplayNormalVisitor.js

```diff
@@ -51,7 +51,7 @@ CompilerOffsetNormal.prototype = MACROUTILS.objectInherit( Compiler.prototype, {
         vertexOffset = this.createVariable( 'vec3', 'vertexOffset' );
 
         // normalize len (divide scale of worldmat, don't work for non uniform scale, but work enough for debug)
-        var str = '%out = %offset == 1.0 ? %vertex + normalize(%vecOffset.xyz) * %scale / %world[0][0] : %vertex;';
+        var str = '%out = %offset == 1.0 ? %vertex + normalize(%vecOffset.xyz) * %scale / length(%world[0].xyz) : %vertex;';
```

Comment:
Nice

"vecOffset" parameter name is a bit misleading though if it's a direction we do normalize to compute the offset ?
Why not "vecOffsetDirection" or even directly "direction" ?
(vecOffset sounds like the result we want to get)

For non unif scale why not do that in world space ? ()
```
var str = '%out = %offset == 1.0 ? %world*vec4(%vertex.xyz,1.0) + normalize(world*vec4(%vecOffset.xyz,0.0)) * %scale / %vertex;';
```
and then using the out to compute View*vertexWorld instead of ModelView*vertex